packages/core/src/endowmentsToolkit.js

```diff
@@ -126,14 +141,49 @@ function endowmentsToolkit({
     )
     // sort by length to optimize further steps
     whitelistedReads.sort((a, b) => a.length - b.length)
-    return makeMinimalViewOfRef(
+    const endowments = makeMinimalViewOfRef(
       sourceRef,
       whitelistedReads,
       unwrapTo,
       unwrapFrom,
       explicitlyBanned,
       allowedWriteFields
     )
+
+    if (allowRedefine.length > 0 && unwrapFrom) {
+      const expectedTarget = unwrapFrom || endowments
+      const originalDefineProperty = unwrapFrom.Object.defineProperty
+      /**
+       * @param {any} target
+       * @param {PropertyKey} key
+       * @param {PropertyDescriptor} descriptor
+       * @returns {boolean}
+       */
+      const redefine = (target, key, descriptor) => {
+        if (
+          target === expectedTarget &&
+          typeof key !== 'symbol' &&
+          allowRedefine.includes(`${key}`)
+        ) {
+          target[key] = descriptor.value
+          return true
+        }
+        originalDefineProperty(target, key, descriptor)
+        return true
+      }
+      endowments.Object = assign(
```

Comment:
the return type of `makeMinimalViewOfRef` should include `Object?: U` (where `U` is the type of whatever you're assigning to it here). which probably means that the type argument `T` of `makeMinimalViewOfRef` extends `Object?: U`

if you don't want to deal with this (I don't blame you), I can just push a commit addressing the typing issues myself after you're satisfied with everything else.

Reply:
I'm not yet sure this is the way to go. But I don't see nicer alternatives

Reply:
```ts
interface SomeGlobalObj = {
  Object: typeof Object;
  [x: string]: unknown;
}
```